### PR TITLE
Update cardano-ledger-specs dependency

### PR DESCRIPTION
**Pull request by nfrisby**
This is a single commit. See the message.

### CODE REVIEW

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs
Comment (Jasagredo) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs:39]: `Cardano.Ledger.Era` was qualified above as `Core`. Maybe we should keep the same consistent naming, but also it could be the case that this module specifically deals with stuff very specific to `Cardano.Ledger.Era`. In that case I would maybe qualify it as `Era`?

**Review by Jasagredo — APPROVED** [review]
Except from that optional comment, the rest looks fine. However I did not look into the non-edited files to search for more occurrences of similar stuff. Maybe it would be worth modifying other files to make the qualification more uniform across our code.

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs
Comment (nfrisby) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs:39]: Ah; I have commit on the next PR that does that. I'll move it to this one.

I went with `.Era` because generally what we're getting from there is the same conceptual thing as we get from `.Core` --- even the module header on `.Era` says it exports "core" stuff.

Comment (nfrisby) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs:39]: > Maybe it would be worth modifying other files to make the qualification more uniform across our code.

That is for Issue #3105, which I' don't want to address yet. I needed to rename them in the modules in this PR to avoid import conflicts, so I did it where necessary. There are several other modules with out-of-date imports.

We'll need to do a broader review and pick a Pareto-optimal scheme -- there are unfortunately multiple dimensions that come to mind.

**Comment by nfrisby** [comment]
I pushed up a commit that disables the failing tests. See Issue #3134.

However, we might be able to fix up ledger in the next couple hours. So I'll wait until then -- maybe we can merge without this workaround.

**Comment by nfrisby** [comment]
@Jasagredo I re-requested review, because I added the `Coherent` newtype and adjusted some instances.

The new `Arbitrary (Coherent (ShelleyBlock era))` instance in `Test.Consensus.Shelley.Generators` is the key change. Jared rapidly added a new generator to `cardano-ledger-specs` that is more careful than the (recently changed) `Arbitrary (SL.Block era)` instance that has supported our existing `Arbitrary (ShelleyBlock era)` instance. So now the properties we have that actually expect some integrity to the arbitrary Shelley blocks explicitly requires it via the `Coherent` wrapper in their arguments. (A `suchThat` or QC implication would have been hopelessly improbable.)

**Review by Jasagredo — APPROVED** [review]
Looks reasonable. `Coherent` looks right and the uses seem correct. I'm fine with merging this.

**Comment by Jasagredo** [comment]
The Hydra build errors seem to be due to the long compiling times on `Ouroboros.Consensus.Cardano.Node`. Perhaps we should turn off those optimizations while this bug is present.

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs
Comment (Jasagredo) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Eras.hs]: Perfect! resolving this thread then.

**Review by EncodePanda — APPROVED** [review]
Thank you for well described commit message. It was straight-forward to do the review (even after 3 days-off :) ).
In general I under the need for the `Coherent` wrapper. I get why some instances were wrapped, though I'm not sure why some other instances were wrapped. But I assume that was because GHC required it :)

File: ouroboros-consensus-mock-test/src/Test/Consensus/Ledger/Mock/Generators.hs
Comment (nfrisby) [reply ouroboros-consensus-mock-test/src/Test/Consensus/Ledger/Mock/Generators.hs:75]: I updated the comment a bit and added a corresponding one on the other instance. Hope that helps.

**Comment by nfrisby** [comment]
b117726b4 is b29f6e190 with comments/stylish-haskell changes only.

**Comment by nfrisby** [comment]
I created a local variant that removes `Coherent` et al and only uses `genCoherentBlock` for `Arbitrary (ShelleyBlock era)`. I built that and am running the tests suites again locally. If that has roughly the same times as b117726 , then I'll merge this PR as-is. If not, I'll force-push the one without `Coherent` and merge that.

**Comment by nfrisby** [comment]
The timings seem the same. Merging.

**Comment by nfrisby** [comment]
1d5b0c7a5 is b117726 rebased onto `origin/master`

**Pull request merged**
